pkg/certificate/providers/vault/certificate_manager.go

```diff
@@ -30,72 +25,36 @@ const (
 	commonNameField   = "common_name"
 	ttlField          = "ttl"
 
-	checkCertificateExpirationInterval = 5 * time.Second
-	decade                             = 8765 * time.Hour
+	decade = 8765 * time.Hour
 )
 
-// NewCertManager implements certificate.Manager and wraps a Hashi Vault with methods to allow easy certificate issuance.
-func NewCertManager(
-	vaultAddr,
-	token string,
-	role string,
-	cfg configurator.Configurator,
-	serviceCertValidityDuration time.Duration,
-	msgBroker *messaging.Broker) (*CertManager, error) {
+// New will construct a new certificate client using Vault's cert-manager
+func New(vaultAddr, token, role string) (*CertManager, error) {
+	if role == "" {
+		return nil, errors.Errorf("Error creating Vault CertManager, expected role got %s", role)
+	}
+
 	c := &CertManager{
-		role:                        vaultRole(role),
-		cfg:                         cfg,
-		serviceCertValidityDuration: serviceCertValidityDuration,
-		msgBroker:                   msgBroker,
+		role: role,
 	}
 	config := api.DefaultConfig()
 	config.Address = vaultAddr
 
 	var err error
 	if c.client, err = api.NewClient(config); err != nil {
-		return nil, errors.Errorf("Error creating Vault CertManager without TLS at %s", vaultAddr)
+		return nil, errors.Errorf("Error creating Vault CertManager without TLS at %s", c.client.Address())
```

Comment:
this could result in nil pointer dereference if an err is returned and client is nil, this should be vaultAddr